controllers/apiserver.go

```diff
@@ -86,3 +87,47 @@ func (r *DSPAReconciler) ReconcileAPIServer(ctx context.Context, dsp *dspav1alph
 	log.Info("Finished applying APIServer Resources")
 	return nil
 }
+
+func (r *DSPAReconciler) GetAPIServerServiceHostname(ctx context.Context, dsp *dspav1alpha1.DataSciencePipelinesApplication) (string, error) {
+	service := &corev1.Service{}
+	namespacedNamed := types.NamespacedName{Name: "ds-pipeline-" + dsp.Name, Namespace: dsp.Namespace}
+	err := r.Get(ctx, namespacedNamed, service)
+	if err != nil {
+		return "", err
+	}
+
+	// Loop over all Service ports, if a secured port is found
+	// set port and scheme to its secured ones and skip the loop
+	serviceScheme := ""
+	servicePort := ""
+	for i := 0; i < len(service.Spec.Ports); i++ {
```

Comment:
it's idiomatic to use range here

```suggestion
	for port := range service.Spec.Ports {
```